crates/router/src/connector/trustpay/transformers.rs

```diff
@@ -0,0 +1,964 @@
+                    message: msg.unwrap_or_default(),
+                    reason: None,
+                    status_code,
+                })
+            } else {
+                None
+            };
+            let payment_response_data = types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(instance_id),
+                redirection_data,
+                mandate_reference: None,
+                connector_metadata: None,
+            };
+            Ok((status, error, payment_response_data))
+        }
+        (_, _, Some(payment_request_id), Some(gateway_url), _, _) => {
+            let status = enums::AttemptStatus::AuthenticationPending;
+            let error = None;
+            let form_fields: HashMap<String, String> = HashMap::from_iter(
+                gateway_url
+                    .query_pairs()
+                    .map(|(key, value)| (key.to_string(), value.to_string())),
+            );
+            let payment_response_data = types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(
+                    payment_request_id.to_string(),
+                ),
+                redirection_data: Some(services::RedirectForm {
+                    endpoint: gateway_url.to_string(),
+                    method: services::Method::Get,
+                    form_fields,
+                }),
+                mandate_reference: None,
+                connector_metadata: None,
+            };
+            Ok((status, error, payment_response_data))
+        }
+        (_, _, _, _, Some(result_info), _) => {
+            let status = enums::AttemptStatus::AuthorizationFailed;
+            let error = Some(types::ErrorResponse {
+                code: result_info.result_code.to_string(),
+                message: result_info.additional_info.unwrap_or_default(),
+                reason: None,
+                status_code,
+            });
+            let payment_response_data = types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::NoResponseId,
+                redirection_data: None,
+                mandate_reference: None,
+                connector_metadata: None,
+            };
+            Ok((status, error, payment_response_data))
+        }
+        (_, _, _, _, _, Some(status)) => {
+            let status = enums::AttemptStatus::from(status);
+            let error = if status == enums::AttemptStatus::AuthenticationFailed {
+                let reason_info = response
+                    .payment_information_response
+                    .unwrap_or_default()
+                    .status_reason
+                    .unwrap_or_default();
+                Some(types::ErrorResponse {
+                    code: reason_info.reason.code,
+                    message: reason_info.reason.reject_reason.unwrap_or_default(),
```

Comment:
do not use default empty value here, use `consts::NO_ERROR_MESSAGE` and `consts::NO_ERROR_CODE`

Reply:
Done